sktree/datasets/hyppo.py

```diff
@@ -135,49 +180,228 @@ def make_trunk_classification(
+                for i in mixture_idx
+            ),
+            dtype=np.dtype((float, n_informative)),
+        )
+
+        X = np.vstack(
+            (
+                rng.multivariate_normal(
+                    np.zeros(n_informative), cov * (2 / 3) ** 2, n_samples // 2, method=method
+                ),
+                X_mixture.reshape(n_samples // 2, n_informative),
+            )
+        )
+    elif simulation in MARRON_WAND_SIMS.keys():
         mixture_idx = rng.choice(
-            [0, 1], n_samples // 2, replace=True, shuffle=True, p=[mix, 1 - mix]
+            len(MARRON_WAND_SIMS[simulation]),  # type: ignore
+            size=n_samples // 2,
+            replace=True,
+            p=MARRON_WAND_SIMS[simulation],
+        )
+        norm_params = MarronWandSims(n_dim=n_informative, cov=cov)(simulation)
+        G = np.fromiter(
+            (
+                rng.multivariate_normal(*(norm_params[i]), size=1, method=method)
+                for i in mixture_idx
+            ),
+            dtype=np.dtype((float, n_informative)),
         )
-        X_mixture = np.zeros((n_samples // 2, len(mu_1)))
-        for idx in range(n_samples // 2):
-            if mixture_idx[idx] == 1:
-                X_sample = rng.multivariate_normal(mu_1, cov, 1, method=method)
-            else:
-                X_sample = rng.multivariate_normal(mu_0, cov, 1, method=method)
-            X_mixture[idx, :] = X_sample
 
+        # as the dimensionality of the simulations increasing, we are adding more and
+        # more noise to the data using the w parameter
+        w = mu_1
         X = np.vstack(
             (
                 rng.multivariate_normal(
                     np.zeros(n_informative), cov, n_samples // 2, method=method
                 ),
-                X_mixture,
+                (1 - w)
+                * rng.multivariate_normal(
+                    np.zeros(n_informative), cov, n_samples // 2, method=method
+                )
+                + w * G.reshape(n_samples // 2, n_informative),
             )
         )
+    else:
+        raise ValueError(
+            f"Simulation must be: trunk, trunk_overlap, trunk_mix, {MARRON_WAND_SIMS.keys()}"
+        )
 
     if n_dim > n_informative:
         X = np.hstack((X, rng.uniform(low=0, high=1, size=(X.shape[0], n_dim - n_informative))))
 
     y = np.concatenate((np.zeros(n_samples // 2), np.ones(n_samples // 2)))
 
     if return_params:
-        return X, y, [mu_0, mu_1], [cov, cov]
+        returns = [X, y]
+        if simulation == "trunk":
+            returns += [[mu_0, mu_1], [cov, cov]]
+        elif simulation == "trunk-overlap":
+            returns += [[np.zeros(n_informative), np.zeros(n_informative)], [cov, cov]]
+        elif simulation == "trunk-overlap":
+            returns += [*list(zip(*norm_params)), X_mixture]
+        else:
+            returns += [*list(zip(*norm_params)), G, w]
```

Comment:
What are `norm_params` exactly? Is it possible to describe where they are initialized what they are? Just because otw it's hard to parse.